api/controllers/UserController.js

```diff
@@ -90,6 +90,32 @@ module.exports = {
     return res.ok();
   },
 
+  submitEmail: async function (req, res) {
+    try {
+      const { username, email } = req.body;
+      const updatedUser = await User.updateOne({ username: username }).set({ email: email });
```

Comment:
This makes sense except that we should also verify that the requesting user is the one specified by the username. Otherwise anyone could request to change anyone else's email with an ajax request. This boils down to looking up the user based on `req.session.usr` and then verifying that the user you get back has the same username as the requesting one
```suggestion
      const requestingUser = await User.findOne({ id: req.session.usr });
      if (requestingUser.username !== username) {
        return res.forbidden({message: 'You can only change your own email address'});
      }
      const updatedUser = await User.updateOne({ username: username }).set({ email: email });
```
A couple notes on how this works
1. sails exposes `req.session` as a readable and writeable object that is unique to the browser window making the request
2. When you log in, the application sets the 'usr' attribute of the session to your user id. This is how we know who you are.
3. We should additionally use policies to ensure that anyone making this request is logged in and has provided a valid email address

Reply:
Here's the sails [documentation on policies](https://sailsjs.com/documentation/concepts/policies#policies). The basic idea is that a policy is a middleware function that runs before a controller action does, usually to validate whether it's allowed or not. They have essentially 2 parts:

1. We'll define a new policy in `api/policies/hasValidEmail.js` which could use the same regex you're using on the frontend to validate that `req.body` has an `email` attribute and that the email and return a `res.badRequest()` if its missing or invalid
2. Then update `config/policies.js` to map this new endpoint to two policies: `'isLoggedIn'` and the new `'hasValidEmail'`
That way without complicating the controller logic itself we can ensure that this action only runs if the user is logged in and if they've provided a valid email